src/functions/module_array.ts

```diff
@@ -0,0 +1,826 @@
+    return [result];
+  },
+  isExported: true,
+};
+
+// -----------------------------------------------------------------------------
+// HSTACK
+// -----------------------------------------------------------------------------
+export const HSTACK: AddFunctionDescription = {
+  description: _lt("Appends ranges horizontally and in sequence to return a larger array."),
+  args: [
+    arg("range1 (any, range<any>)", _lt("The first range to be appended.")),
+    arg("range2 (any, range<any>, repeating)", _lt("Additional ranges to add to range1.")),
+  ],
+  returns: ["RANGE<ANY>"],
+  //TODO computeFormat
+  compute: function (...ranges: ArgValue[]): CellValue[][] {
+    const _ranges = ranges.map((range) => toMatrixArgValue(range));
+
+    const nRows = Math.max(..._ranges.map((range) => range[0].length));
+
+    const colArray: CellValue[][] = [];
+    for (const range of _ranges) {
+      for (const col of range) {
+        //TODO: fill with #N/A for unavailable values instead of zeroes
+        const paddedCol: CellValue[] = Array(nRows).fill(0);
+        for (let i = 0; i < col.length; i++) {
+          paddedCol[i] = toCellValue(col[i]);
+        }
+        colArray.push(paddedCol);
+      }
+    }
+    return colArray;
+  },
+  isExported: true,
+};
+
+// -----------------------------------------------------------------------------
+// MDETERM
+// -----------------------------------------------------------------------------
+export const MDETERM: AddFunctionDescription = {
+  description: _lt("Returns the matrix determinant of a square matrix."),
+  args: [
+    arg(
+      "square_matrix (number, range<number>)",
+      _lt(
+        "An range with an equal number of rows and columns representing a matrix whose determinant will be calculated."
+      )
+    ),
+  ],
+  returns: ["NUMBER"],
+  compute: function (matrix: ArgValue): number {
+    if (!isMatrix(matrix)) {
+      return toNumber(matrix);
+    }
+    const _matrix: Matrix<number> = matrixMap(matrix, toNumber);
+
+    assertSquareMatrix(
+      _lt("The argument square_matrix must have the same number of columns and rows."),
+      _matrix
+    );
+
+    const { determinant } = invertMatrix(_matrix);
+
+    return determinant;
+  },
+  isExported: true,
+};
+
+// -----------------------------------------------------------------------------
+// MINVERSE
+// -----------------------------------------------------------------------------
+export const MINVERSE: AddFunctionDescription = {
+  description: _lt("Returns the multiplicative inverse of a square matrix."),
+  args: [
+    arg(
+      "square_matrix (number, range<number>)",
+      _lt(
+        "An range with an equal number of rows and columns representing a matrix whose multiplicative inverse will be calculated."
+      )
+    ),
+  ],
+  returns: ["RANGE<NUMBER>"],
+  compute: function (matrix: ArgValue): CellValue[][] {
+    const _matrix: Matrix<number> = matrixMap(toMatrixArgValue(matrix), toNumber);
```

Comment:
When do you need the `toMatrixArgValue` ? Is it only when the user enter a number as argument directly, or also when targeting a range with only a cell (I think in this last case we still have a matrix but not sure) ?